test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs

```diff
@@ -151,20 +150,17 @@ public class OpenIdConnectMiddlewareTests
 
         private static void SetProtocolMessageOptions(OpenIdConnectOptions options)
         {
-            var mockOpenIdConnectMessage = new Mock<OpenIdConnectMessage>();
-            mockOpenIdConnectMessage.Setup(m => m.CreateAuthenticationRequestUrl()).Returns(ExpectedAuthorizeRequest);
-            mockOpenIdConnectMessage.Setup(m => m.CreateLogoutRequestUrl()).Returns(ExpectedLogoutRequest);
             options.AutomaticAuthentication = true;
             options.Events = new OpenIdConnectEvents()
             {
                 OnRedirectToAuthenticationEndpoint = (context) =>
                 {
-                    context.ProtocolMessage = mockOpenIdConnectMessage.Object;
+                    context.ProtocolMessage = new OpenIdConnectMessage(ExpectedAuthorizeRequest);
```

Comment:
I'm not sure how that message type works, but this certainly *looks* like it's different behavior. The mock was returning these request objects only when certain methods were called.